sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1611,7 +1611,9 @@ def _retry_zones(
             global_user_state.set_owner_identity_for_cluster(
```

Comment:
We had some optimization in `_update_blocklist_on_gcp_error` to block out cloud, region or zone based on the error output. However, the current implementation only blocks out the zone or region specified. Is it possible to make it error specific? We can even refactor that `_update_blocklist_on_gcp_error` function to new provisioner so that the provisioner API can decide what resources to block based on the errors.

Reply:
fixed